### DISCUSSION TITLE
raspberry pi 3b or 4?

### DISCUSSION THREAD

**Discussion by devinesteve** [artisan-roaster-scope/artisan#957] [Q&A] [answered]
I am getting ready to buy a Raspberry Pi - I see that the software packages for Artisan are listed as Raspberry Pi 3b compatible. It seems as if Raspberry Pi 3b is fairly old and I would prefer buy the 4 to be on the safe side.
So my question -- will the Artisan software packages run on Raspberry Pi 4 - does anyone have any experience with this? 

Thanks in advance
Steve

**Comment by bigl00z3** [comment, marked as answer]
Hello Steve,

I have been using Artisan on Raspberry Pi now for years. Initially with the 3b and now for a few years on the 4. **Very happy with it.**
My setup is using a bunch of Phidgets and also Modbus communication to read and control a Fuji PXF PID.
Of course a Raspberry is a little bit slower than a full size computer with high spec but for the price difference it works very well. 
Just good to know that they are less frequent continuous releases available.

Nico

**Reply by devinesteve** [reply]
Thanks, I will order one today.

**Reply by N0YHR** [reply]
I've been roasting for twelve years and initially wrote a data logger and charting program in Excel vba.  I used an Arduino Duemilanov, a K-Type thermocouple, and an air flow sensor.  It worked great till Office was upgraded from 32 bit to 64, and some libraries were abandoned.

I just saw Artisan yesterday and while I still have a roaster that cannot be controlled; Freshroast SR800, I would like to record termperature data with a Raspberry Pi running the latest 64 bit version of Raspberry Pi OS.  When setting up Artisan, what hardware should I select?  I don't see Arduinos or Raspberry Pis as options.

**Reply by N0YHR** [reply]
Thank you for your response.  Under Config/Machine, what machine should I select as I will only be using the Pi and thermocouples?

**Reply by poundy** [reply]
The Config-Machine menu option just loads pre-set parameters into Artisan for machines that are known by the Artisan Project - your scenario means you'll do this all manually.

Artisan requires a "driver" to talk to the device that reads thermocouples. In your case, you have a generic serial device (your arduino) but how you wrote the arduino sketch could be very different to how I would.  To get around this, Artisan implements known communication protocols - the simplest one for your scenario is the TC4. You can set Artisan up with a TC4 device, and load a TC4 (or compatible) firmware to your arduino, and things will just work (I previously used a micro and wrote a TC4 compatible firmware for that, as well as displaying details on a 4.3" LCD).  If you didn't want to go down that path but retain your own Arduino code, you can write a script to sit between the Artisan layer and your serial port.

**Reply by N0YHR** [reply]
Thank you very much for responding.  I’ve looked at the TC4+ and the Phidgets and will likely go that direction using the scripts that are available.

**Comment by MAKOMO** [comment]
Raspberry PI OS 64bit is not supported. However, Raspberry Pi OS 32bit (Raspbian) is.  Choose ```artisan-linux-x.x.x_raspbian-bullseye.deb```
